components/calendar/calendar.js

```diff
@@ -325,57 +331,66 @@ class Calendar extends LocalizeStaticMixin(LitElement) {
 				};
-				const focused = checkIfDatesEqual(day, this._focusDate);
+				const year = day.getFullYear();
+				const month = day.getMonth();
+				const date = day.getDate();
 				return html`
 					<td>
 						<div
+							aria-label="${formatDate(day, {format: 'full'})}"
+							aria-selected="${selected}"
 							@click="${this._onDateSelected}"
 							@keydown="${this._onKeyDown}"
 							class=${classMap(classes)}
-							data-date=${day.getDate()}
-							data-month=${day.getMonth()}
-							data-year=${day.getFullYear()}
+							data-date=${date}
+							data-month=${month}
+							data-year=${year}
+							headers="${this._tableInfoId}-weekday-${index}"
+							id="${this._tableInfoId}-${year}-${month}-${date}"
+							role="gridcell"
 							tabindex=${focused ? '0' : '-1'}>
-							${day.getDate()}
+							${date}
 						</div>
 					</td>`;
 			});
 
-			return html`<tr>${weekHtml}</tr>`;
+			return html`<tr role="row">${weekHtml}</tr>`;
 		});
+		const activeDate = `${this._tableInfoId}-${this._focusDate.getFullYear()}-${this._focusDate.getMonth()}-${this._focusDate.getDate()}`;
 		const heading = `${calendarData.descriptor.calendar.months.long[this._shownMonth]} ${this._shownYear}`;
 		return html`
-			<div aria-labelledby="${this._labelId}" class="d2l-calendar">
-				<table summary="${this.summary}" role="grid">
-					<thead>
-						<tr class="d2l-calendar-title">
-							<td>
-								<d2l-button-icon
-									@click="${this._onPrevMonthButtonClick}"
-									text="${this._computeText(getPrevMonth(this._shownMonth))}"
-									icon="tier1:chevron-left">
-								</d2l-button-icon>
-							</td>
-							<th colspan="5">
-								<h2 class="d2l-heading-4" aria-live="polite" id="${this._labelId}">${heading}</h2>
-							</th>
-							<td>
-								<d2l-button-icon
-									@click="${this._onNextMonthButtonClick}"
-									text="${this._computeText(getNextMonth(this._shownMonth))}"
-									icon="tier1:chevron-right">
-								</d2l-button-icon>
-							</td>
-						</tr>
-						<tr>${weekdayHeaders}</tr>
+			<div aria-labelledby="${this._tableInfoId}-heading" class="d2l-calendar">
+				<div class="d2l-calendar-title">
+					<d2l-button-icon
+						@click="${this._onPrevMonthButtonClick}"
+						text="${this._computeText(getPrevMonth(this._shownMonth))}"
+						icon="tier1:chevron-left">
+					</d2l-button-icon>
+					<h2 aria-atomic="true" aria-live="polite" class="d2l-heading-4" id="${this._tableInfoId}-heading">${heading}</h2>
+					<d2l-button-icon
+						@click="${this._onNextMonthButtonClick}"
+						text="${this._computeText(getNextMonth(this._shownMonth))}"
+						icon="tier1:chevron-right">
+					</d2l-button-icon>
+				</div>
+				<table
+					aria-activedescendant="${activeDate}"
+					aria-labelledby="${this._tableInfoId}-heading"
+					aria-readonly="true"
+					role="grid"
+					summary="${ifDefined(this.summary)}">
+					<thead aria-hidden="false" role="presentation">
+						<tr role="row">${weekdayHeaders}</tr>
```

Comment:
I'm surprised these `tr`s and the `th`s inside them needed row/gridheader roles... is it because of the `grid` role on the table itself that's turning it into something other than a table? If so, I think I'd expect the `td`s to need `presentation` roles too.

Reply:
After some experimenting it seems the best option is to remove the `row` and `presentation` roles. They don't seem to improve anything and different combinations cause different/undesired behavior in some browsers (most problematic seemed to be chrome + voice over)